### PR TITLE
Fixed #32107 -- Fixed ProtectedError.protected_objects and RestrictedError.restricted_objects.

### PR DESCRIPTION
**Pull request by hramezani**
https://code.djangoproject.com/ticket/32107

### CODE REVIEW

File: tests/delete/tests.py [hunk 1]
```diff
@@ -76,18 +76,26 @@ def test_protect(self):
             "referenced through protected foreign keys: 'A.protect'."
         )
         with self.assertRaisesMessage(ProtectedError, msg):
-            a.protect.delete()
+            try:
+                a.protect.delete()
+            except ProtectedError as exc:
+                self.assertEqual(exc.protected_objects, [a])
+                raise
```
Comment (felixxm) [inline tests/delete/tests.py:83]: You don't need to catch and reraise the exception:
```python
        with self.assertRaisesMessage(ProtectedError, msg) as cm:
            a.protect.delete()
        self.assertEqual(cm.exception.protected_objects, [a])
```

Comment (hramezani) [reply tests/delete/tests.py]: Thanks @felixxm

**Review by felixxm — COMMENTED** [review]
@hramezani Thanks :+1: I think we should return `set()` instead of `list()`, as we did before 4ca5c565f4dc9e97845036e86416abc5cfde766c.

**Comment by felixxm** [comment]
@hramezani Thanks :+1: I added release notes.

**Pull request merged**